### PR TITLE
feat: [CDS-49237]: Add coverage for freeze notification helper

### PR DESCRIPTION
**Pull request by Y0709**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42999)

### CODE REVIEW

File: .gitleaksignore [hunk 1]
```diff
@@ -2637,4 +2637,5 @@ a140c795626b09be64c1c2b8d28bf4e987bd1521:src/test/java/software/wings/integratio
 0374a2529b1c8f26d96eacf2deb11e50116c15fb:src/test/java/software/wings/integration/DataGenUtil.java:private-key:719
 2317f3d1f1f6d357e77ed4bfdec83ef686cf39c2:src/test/java/software/wings/sm/states/ElasticLoadBalancerStatesTest.java:aws-access-token:36
 2317f3d1f1f6d357e77ed4bfdec83ef686cf39c2:src/test/java/software/wings/sm/states/ElasticLoadBalancerStatesTest.java:aws-access-token:41
-2bac5668076eff169fa8a4cb4a0cf273ee288dbd:930-delegate-tasks/src/test/java/io/harness/delegate/task/cf/artifact/TasElasticContainerRegistrySettingsProviderTest.java:generic-api-key:53
```
Comment (pranay-harness) [inline .gitleaksignore:2640]: Approved. Secret is a base64 encoded token with a dummy value "aws:test-token:ecr"

**Review by pranay-harness — DISMISSED** [review]

File: .gitleaksignore [hunk 2]
```diff
@@ -2637,4 +2637,5 @@ a140c795626b09be64c1c2b8d28bf4e987bd1521:src/test/java/software/wings/integratio
...
+db46bee78809147678c3191373a3216c644e4533:125-freeze/src/test/resources/FreezeWithoutNotifications.yaml:slack-web-hook:58
```
[the remark this answers is not in the document]
Comment (Y0709) [reply .gitleaksignore:2640]: Just for testing . Will not be making a real request. The url is also invalid

**Review by Abhishek-CDC — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42999)* status: 4 file(s) to review. 1. unresolved discussions. 1 of 2 approvals obtained.

**Review by jyotiraval — APPROVED** [review]
The url used is incorrect used the yaml for a unit test which wont make a real request to slack

**Review by abhinavmittal-harness — APPROVED** [review]
> Reviewed 1 of 3 files at r1, 2 of 3 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42999)* status: 1. unresolved discussions.

**Review by abhinavmittal-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42999)* status: 1. unresolved discussions.

**Review by abhinavmittal-harness — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42999)* status: 1. unresolved discussions.

**Review by pranay-harness — APPROVED** [review]

**Pull request merged**